<a href="https://www.kaggle.com/code/hassaneskikri/chi-square-statistical-tests?scriptVersionId=168275828" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
%%html
<style>

* {
    font-size: 18px;
    line-height: 1.5;
    font-family: 'Arial', sans-serif;
    align-item : center;
    justifiy-content:center;
    max-width : 1000px;

}

img{
    display: flex;
    margin-left: auto;
    margin-right: auto;
    width: 700px;
    height: auto;
    text-align: center;
    border-radius: 15px;
}
    
h1,
h2 {
  font-family: Impact, Charcoal, sans-serif;
  font-weight: bold;
  text-shadow: 2px 2px 4px #000000;
}

h1 {
  font-size: 52px;
  margin-bottom: 40px;
  color: #10929e;
  text-decoration: underline;
    text-align:center;
}

h2 {
  font-size: 448px;
  margin-bottom: 32px;
  color: #76b4be;
  text-transform: uppercase;
}

/* Block Quotes */
blockquote {
  font-family: Georgia, serif;
  font-size: 16px;
  color: #19085c;
  border-left: 8px solid #effffc;
  background-color: #eafcff;
  padding: 20px;
}
    
</style>

A Pearson’s chi-square test is a statistical test for categorical data. It is used to determine whether your data are significantly different from what you expected. There are two types of Pearson’s chi-square tests:

- The chi-square goodness of fit test is used to test whether the frequency distribution of a categorical variable is different from your expectations.
- The chi-square test of independence is used to test whether two categorical variables are related to each other.

Chi-square is often written as Χ2 and is pronounced “kai-square” (rhymes with “eye-square”). It is also called chi-squared.


The chi-square test is a statistical method used to compare observed frequencies of categorical variables against expected frequencies. It's particularly useful when you're dealing with nonparametric data, meaning the data doesn't necessarily follow a normal distribution.

# What It Is:

- Nonparametric Test: Ideal for data that don't fit the normal distribution assumption.
- Categorical Variables: Used for data categorized into groups, like types of animals or different brands.

# Types of Chi-Square Tests:

- Goodness of Fit Test: Checks if a single categorical variable follows a specific distribution.
  * Example: Testing if different bird species visit a feeder in equal proportions.
- Test of Independence: Determines if there's a relationship between two categorical variables.
  * Example: Seeing if handedness (left or right) is related to nationality (American or Canadian).

# When to Use It:

- You're working with categorical data.
- Your sample is randomly selected.
- Each category or combination of categories has at least five observations.

# How It Works:
- Calculate Expected Frequencies: Based on your hypothesis, determine what the frequencies of each category should be if there were no real effect or difference.
- Observe Actual Frequencies: Count how many times each category actually occurs in your data.
- Chi-Square Formula: Use the formula Χ2 = Σ((O-E)^2/E) where O is observed frequency and E is expected frequency. This calculates the chi-square statistic, which measures the difference between your observed and expected frequencies.
- Compare to Critical Value: Determine if the chi-square statistic is large enough to indicate a statistically significant difference between observed and expected frequencies.

# Reporting Results:

- Mention the chi-square statistic (Χ2), degrees of freedom (based on the number of categories minus one), the sample size (N), and the p-value (which tells you if the result is statistically significant).

# Simple Summary:

The chi-square test helps you understand if the differences you see in categorical data (like types of birds at a feeder or handedness among people from different countries) are likely due to chance or if they're statistically significant. It's a way to test hypotheses about how data is distributed across different categories or if two variables are related to each other.

# Chi-Square Goodness of Fit Test

The **Chi-Square (Χ2) Goodness of Fit Test** is a statistical method used to determine whether the observed distribution of a categorical variable differs significantly from an expected distribution.

## Key Points

- **Purpose**: To test if the observed distribution of a single categorical variable matches the expected distribution.
- **Type of Test**: Nonparametric, used for categorical variables.
- **Applications**: Commonly used in genetics, marketing research, and other fields to analyze preferences or natural occurrences.

## Hypotheses

- **Null Hypothesis (H0)**: The population follows the specified distribution.
- **Alternative Hypothesis (Ha)**: The population does not follow the specified distribution.

## Conditions for Use

1. The variable in question is categorical.
2. The sample is randomly selected.
3. Each category is expected to have a minimum of five observations.

## Calculation Steps

1. **Expected Frequencies**: Determine the expected frequencies based on the hypothesis.
2. **Chi-Square Statistic**: Calculate using the formula `Χ2 = Σ((O-E)^2/E)`, where `O` is observed frequency and `E` is expected frequency.
3. **Critical Value**: Find the critical chi-square value from a table or software, based on degrees of freedom (`df = number of categories - 1`) and significance level (usually `α = .05`).
4. **Comparison**: Compare the chi-square statistic to the critical value to decide on the null hypothesis.
5. **Decision**: Reject the null hypothesis if the chi-square statistic is greater than the critical value, indicating a significant difference between observed and expected frequencies.

## Example

A dog food company tests three new flavors with 75 dogs, expecting an equal preference. The observed preferences slightly differ from the expected equal distribution. A chi-square goodness of fit test is used to determine if the differences are statistically significant.

## When to Use a Different Test

- **Chi-Square Test of Independence**: When you have two categorical variables and want to test their relationship.
- **Anderson–Darling or Kolmogorov–Smirnov Tests**: For continuous variables or when a more specific goodness of fit test is applicable to the distribution of interest.

## Conclusion

The chi-square goodness of fit test is a valuable tool for analyzing how well observed categorical data matches an expected distribution. It helps researchers and professionals make informed decisions based on statistical evidence.


# Chi-Square Test of Independence

The **Chi-Square (Χ2) Test of Independence** is a nonparametric hypothesis test used to determine if there is a significant relationship between two categorical variables.

## Example Scenario

A city tests two interventions to encourage recycling among its residents: an educational flyer and a phone call. The city uses a chi-square test of independence to analyze if the intervention type affects recycling behavior.

## What is the Chi-Square Test of Independence?

- A type of Pearson’s chi-square test.
- Determines if two categorical variables are related.
- Based on observed frequencies in a contingency table.

## Hypotheses

- **Null Hypothesis (H0)**: The two variables are not related; proportions of one variable are the same across the values of the other variable.
- **Alternative Hypothesis (Ha)**: The two variables are related; proportions of one variable differ across the values of the other variable.

## When to Use It

- Testing a hypothesis about the relationship between two categorical variables.
- The sample is randomly selected.
- Each combined group is expected to have a minimum of five observations.

## Calculation Steps

1. **Expected Frequencies**: Calculate using the formula: `(Row total × Column total) / Grand total`.
2. **Chi-Square Statistic**: Calculate using Pearson’s chi-square formula: `Χ2 = Σ((O-E)^2/E)` where `O` is the observed frequency and `E` is the expected frequency.

## Performing the Test

1. Calculate expected frequencies using the contingency table.
2. Calculate the chi-square statistic.
3. Find the critical chi-square value based on degrees of freedom and significance level (usually .05).
4. Compare the chi-square value to the critical value to decide on the null hypothesis.
5. If the chi-square value is greater than the critical value, reject the null hypothesis, indicating a significant relationship between the variables.

## Example

After analyzing recycling behavior in 300 households, the city finds a significant difference in recycling rates between the intervention groups, suggesting that the interventions affect recycling behavior.

## When to Use a Different Test

- **Chi-Square Goodness of Fit Test**: When you have one categorical variable and want to test its distribution.
- **Fisher’s Exact Test**: For small sample sizes or when expected frequencies in any group are less than five.
- **McNemar’s Test**: For paired categorical data to compare before and after observations.
- **G Test**: An alternative to chi-square tests for more complex designs, though both tests usually yield similar results.

The choice of test depends on the variables, sample size, and specific hypotheses of your study.


# implemntation python

## Chi-Square Goodness of Fit Test

Imagine a scenario where a dog food company introduces three new flavors (Chicken, Beef, and Vegan) and wants to see if there's an equal preference among 75 dogs.

In [2]:
from scipy.stats import chisquare


observed_frequencies = [25, 30, 20]  # e.g., 25 for Chicken, 30 for Beef, 20 for Vegan

# Expected frequencies if dogs had no preference (equal distribution)
expected_frequencies = [25, 25, 25]  # 75 dogs / 3 flavors


chi_stat, p_value = chisquare(f_obs=observed_frequencies, f_exp=expected_frequencies)

print(f"Chi-Square Goodness of Fit Test results -- Chi-square statistic: {chi_stat}, p-value: {p_value}")


Chi-Square Goodness of Fit Test results -- Chi-square statistic: 2.0, p-value: 0.36787944117144245


The test result (p-value = 0.36787944117144245) suggests that the observed differences in preferences for the three dog food flavors (Chicken, Beef, Vegan) among the 75 dogs are not statistically significant. This means there's no strong evidence to reject the idea that dogs have no preference among these flavors.

## Chi-Square Test of Independence Example

Consider a study comparing two interventions (an educational flyer and a phone call) to encourage recycling among residents. We have data from 300 households on whether they increased their recycling behavior following the intervention.

In [3]:
import numpy as np
from scipy.stats import chi2_contingency

# Observed frequencies in a contingency table
# Rows: Educational Flyer, Phone Call
# Columns: Increased Recycling, Did Not Increase
observed = np.array([[120, 30],  # 120 increased, 30 did not after flyer
                     [80, 70]])  # 80 increased, 70 did not after call

chi_stat, p_value, dof, expected = chi2_contingency(observed)

print(f"Chi-Square Test of Independence results -- Chi-square statistic: {chi_stat}, p-value: {p_value}")


Chi-Square Test of Independence results -- Chi-square statistic: 22.815, p-value: 1.7836812173565438e-06


With a p-value of approximately 1.78e-06, the test indicates a statistically significant difference in recycling behavior between the two groups receiving different interventions (educational flyer vs. phone call). This suggests that the type of intervention (flyer or phone call) has a significant impact on whether households increased their recycling efforts.

# Resources



- [ ouverview](https://www.scribbr.com/statistics/chi-square-tests/)
- [chi-square goodness of fit ](https://www.scribbr.com/statistics/chi-square-goodness-of-fit/)
- [chi-square test of independance](https://www.scribbr.com/statistics/chi-square-test-of-independence/)